### PR TITLE
Login filter and error mapper

### PR DESCRIPTION
**Pull request by rpg2014**

### CODE REVIEW

File: src/main/java/com/rpg2014/Providers/JWTKeyProvider.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    JwkProvider keyProvider = new JwkProviderBuilder("https://cognito-idp.us-east-1.amazonaws.com/us-east-1_mX9fI3lzt/.well-known/jwks.json").cached(true).build();
+    final RSAPrivateKey privateKey = null;
+    final String privateKeyId = "notUsed";
+
+    @Override
+    public RSAPublicKey getPublicKeyById(String s) {
+
+        try {
+            System.out.println(keyProvider.get(s).getType());
```
Comment (spencercarr) [inline src/main/java/com/rpg2014/Providers/JWTKeyProvider.java]: remove this

File: src/main/java/com/rpg2014/Providers/JWTKeyProvider.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            if(keyProvider.get(s).getType() == "RSA" ){
+                RSAPublicKey key = (RSAPublicKey) keyProvider.get(s).getPublicKey();
+                return key;
+            }
+
```
Comment (spencercarr) [inline src/main/java/com/rpg2014/Providers/JWTKeyProvider.java]: remove this

File: src/main/java/com/rpg2014/ServerController.java [hunk 3]
```diff
@@ -25,16 +29,41 @@
...
-        TestSuspended.start();
-        return false;
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+
+    public StartResponse serverStart() {
+        new Thread(()-> {
+            TestSuspended.start();
+        }).start();
+
```
Comment (spencercarr) [inline src/main/java/com/rpg2014/ServerController.java]: delete this line

File: src/main/java/com/rpg2014/filters/RequiresLogin/RequiresLoginFeature.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+public class RequiresLoginFeature implements DynamicFeature {
+
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext featureContext) {
+        RequiresLogin requiresLogin = resourceInfo.getResourceMethod().getAnnotation(RequiresLogin.class);
+        if(requiresLogin ==null){
```
Comment (spencercarr) [inline src/main/java/com/rpg2014/filters/RequiresLogin/RequiresLoginFeature.java]: add a space after the ==

File: src/main/java/com/rpg2014/filters/RequiresLogin/RequiresLoginFilter.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    JWTVerifier verifier = JWT.require(algorithm).withIssuer(USER_POOL).withAudience(COGNITO_APP_ID).withClaim("token_use", "access").acceptLeeway(1).build();
+
+    @Override public void filter(ContainerRequestContext containerRequestContext) throws IOException {
+        String authToken = containerRequestContext.getHeaderString(AUTH_HEADER_NAME);
+        if (authToken == null) {
+            throw new NotAuthorizedException("No Auth Header");
+        }
+        String username = "";
+        try {
+            username =  verifyToken(authToken);
```
Comment (spencercarr) [inline src/main/java/com/rpg2014/filters/RequiresLogin/RequiresLoginFilter.java]: extra space after =

File: src/main/java/com/rpg2014/Providers/JWTKeyProvider.java (hunk 2 above)
Comment (rpg2014) [reply src/main/java/com/rpg2014/Providers/JWTKeyProvider.java]: Remove what?

File: src/main/java/com/rpg2014/model/DetailsResponse.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package com.rpg2014.model;
+
+import lombok.Builder;
+
+@Builder
+public class DetailsResponse {
+
```
Comment (spencercarr) [inline src/main/java/com/rpg2014/model/DetailsResponse.java]: delete this line

**Pull request merged**